File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int cnt=0,l=0,r=k-1;
        int sum=0;
        int n=arr.length;
        double avg;
        for(int i=0;i<k;i++){
            sum+=arr[i];
        }
        if(sum>=threshold*k){
            cnt+=1;
        }
        while(r<n-1){
            sum=sum-arr[l];
            l++;
            r++;
            sum=sum+arr[r];
        
        
        if(sum>=threshold*k){
            cnt+=1;
        }
        }
        return cnt;
    }
}

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        cnt=0
        sum=0
        j=0
        m=k
        for i in range(k):
            sum+=arr[i]
        if(sum/k>=threshold):
                cnt+=1
        while(m<len(arr)):
            sum-=arr[j]
            sum+=arr[m]
            j+=1
            m+=1
            avg=sum/k
            if(avg>=threshold):
                cnt+=1
        return cnt


        